### ISSUE TITLE
Change Request: Allow flat config file at non-standard location

### ISSUE THREAD

**Issue by rhysd** [eslint/eslint#18212]
### ESLint version

v8.57.0

### What problem do you want to solve?

My repository has eslint config file at non-standard location (in `eslint/` subdirectory).

This was allowed by eslint with the legacy config using `--config` option.

```sh
npx eslint --config eslint/eslintrc.json ...
```

However, the `--config` option is only for eslintrc files and does not support flat configs. As far as I checked, there seems no way to specify the flat config file path (please correct me if I'm wrong).

### What do you think is the correct solution?

I'd like `--config` command line option to support flat config files. If it is not possible (since both legacy config and flat configs can use `.js` file extension), adding some new option might be an alternative idea.

```sh
npx eslint --config eslint/config.mjs ...
```

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by mdjermanovic** [comment]
Hi @rhysd, thanks for the issue!

`--config` command line option already supports flat config files, but you'll need to switch ESLint to flat config mode first because legacy config mode is the default in v8.x, and there is no flat config file in the current working directory that would automatically switch ESLint to flat config mode. To do this, you can set environment variable `ESLINT_USE_FLAT_CONFIG` to `true`:

```sh
$ ESLINT_USE_FLAT_CONFIG=true npx eslint --config eslint/config.mjs ...
```

https://eslint.org/docs/latest/use/configure/configuration-files-new#configuration-file-resolution

In ESLint v9, flat config mode will be the default, so there will be no need to set the environment variable.

**Comment by rhysd** [comment]
Oh, thank you for the information. I didn't know eslint switches configuration by the modes. I confirmed `ESLINT_USE_FLAT_CONFIG` worked for my project.
